src/python/pants/pantsd/subsystem/watchman_launcher.py

```diff
@@ -31,6 +31,9 @@ def register_options(cls, register):
       register('--supportdir', advanced=True, default='bin/watchman',
                help='Find watchman binaries under this dir. Used as part of the path to lookup '
                     'the binary with --binary-util-baseurls and --pants-bootstrapdir.')
+      register('--startup-timeout', type=float, advanced=True, default=Watchman.SOCKET_TIMEOUT_SECONDS,
```

Comment:
It seems like it would be reasonable to set the startup timeout to a higher value (ie, your 30) by default.